### PR TITLE
MGMT-17560: Report iSCSI host IP address in inventory

### PR DESCRIPTION
**Pull request by adriengentil**
Report iSCSI host IP address in the inventory, it will be used to
perform validations on service side when the iSCSI disk is selected as
installation disk:
- Check that host IP is not part of the default network interface
- Ensure machine network does not contain the iSCSI host IP address

PRs part of this story:
- Validate iSCSI volume eligibility on hosts: https://github.com/openshift/assisted-service/pull/6434
- Report iSCSI host ip address: https://github.com/openshift/assisted-installer-agent/pull/727
- Configure DHCP on iSCSI network interface: https://github.com/openshift/assisted-service/pull/6602
- Workaround DNS issue on first boot: https://github.com/openshift/assisted-service/pull/6603
- Filter out machine networks: TODO

### CODE REVIEW

File: src/inventory/disks.go [hunk 1]
```diff
@@ -512,6 +518,36 @@ func (d *disks) getBusPath(disks []*block.Disk, index int, busPath string) strin
 	return d.getByPath(busPath)
 }
 
+// getISCSIHostIPAddress retuns the IP address in use to connect on the iSCSI volume
+func (d *disks) getISCSIHostIPAddress(diskName string) string {
```
Comment (AlonaKaplan) [inline src/inventory/disks.go:522]: Will this method still be correct in case of multipath?

Comment (adriengentil) [reply src/inventory/disks.go:522]: multpath is not in the scope of this change, we focus only on iSCSI.

When we will decide that we need to support multipath+iSCSI, we will be able to re-use this method by looking at the iSCSI volumes behind the DM volume, and read the IP for each iSCSI volume.

**Comment by adriengentil** [comment]
/hold
requires https://github.com/openshift/assisted-service/pull/6434 to be merged before, in order to update the API/models

File: src/inventory/disks.go [hunk 2]
```diff
@@ -512,6 +518,36 @@ func (d *disks) getBusPath(disks []*block.Disk, index int, busPath string) strin
...
+
+	// Read IP address
+	hostIPAdressFile := fmt.Sprintf("/sys/class/iscsi_host/%s/ipaddress", splitPath[hostIndex])
+	data, err := d.dependencies.ReadFile(hostIPAdressFile)
+	if err != nil {
+		logrus.WithError(err).Errorf("Failed to read host IP address file for iSCSI device")
+		return ""
+	}
+
+	ipAddress := strings.TrimSpace(string(data))
```
[the remark this answers is not in the document]
Comment (adriengentil) [reply src/inventory/disks.go:547]: in by_path, we get the address of the target, right? Here, I retrieve the IP address used by the host to connect on the target.

[the remark this answers is not in the document]
Comment (linoyaslan) [reply src/inventory/disks.go:547]: Right, I see the difference now:
```
[root@test-infra-cluster-409bd9d3-master-0 ~]# cat /sys/class/iscsi_host/host6/ipaddress 
192.168.145.10
```

Thank you for the clarification!

File: go.mod [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module github.com/openshift/assisted-installer-agent
 
-go 1.20
+go 1.21
```
Comment (eifrach) [inline go.mod:3]: please update the Dockerfile also.

Note, the CI has change and it's using the Dockerfile go version.

File: vendor/modules.txt [hunk 4]
```diff
@@ -1146,7 +1146,7 @@ sigs.k8s.io/yaml/goyaml.v2
 # github.com/opencontainers/runc => github.com/opencontainers/runc v1.1.12
 # github.com/openshift/assisted-service/api => github.com/openshift/assisted-service/api v0.0.0-20240531185134-780deffb6a35
 # github.com/openshift/assisted-service/client => github.com/openshift/assisted-service/client v0.0.0-20240531185134-780deffb6a35
-# github.com/openshift/assisted-service/models => github.com/openshift/assisted-service/models v0.0.0-20240531185134-780deffb6a35
+# github.com/openshift/assisted-service/models => github.com/adriengentil/assisted-service/models v0.0.0-20240612141820-d05f3eb1f803
```
Comment (eifrach) [inline vendor/modules.txt:1149]: i think this is a mistake

File: go.mod (hunk 3 above)
Comment (adriengentil) [reply go.mod:3]: ack, I'll rebase and re-tidy/vendor

File: vendor/modules.txt (hunk 4 above)
Comment (adriengentil) [reply vendor/modules.txt]: Now using openshift/assisted-service

**Comment by eifrach** [comment]
know issue 

/override edge-subsystem-test

**Pull request merged**